kernel/src/process/posix_thread/mod.rs

```diff
@@ -141,21 +141,38 @@ impl PosixThread {
         &self.sig_mask
     }
 
+    /// Returns the pending signal of the thread.
+    ///
+    /// This method will also return the signals that are blocked or ignored.
     pub fn sig_pending(&self) -> SigSet {
-        self.sig_queues.sig_pending()
+        let process = self.process();
+        self.sig_queues.sig_pending() | process.sig_queues().sig_pending()
     }
 
     /// Returns whether the thread has some pending signals
-    /// that are not blocked.
+    /// that are not blocked and not ignored.
     pub fn has_pending(&self) -> bool {
```

Comment:
This method is used in the task look (`create_new_user_task`).

I am not sure why it should exclude ignored signals? I think they should be dequeued in the task loop.

Reply:
It appears that when a thread is blocked in certain syscalls (for example, reading from a pipe), it can still consume signals that are ignored but not blocked. I added a new regression test to cover this case.

To reproduce the behavior, I modified the `has_pending` logic so that it dequeues signals that are ignored but not blocked until it finds the first actual pending signal that is neither blocked nor ignored.